extensions/blocks/social-previews/utils.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+import { has } from 'lodash';
+
+/**
+ * Gets the URL of the media. Tries loading a smaller size available to be used as a thumbnail and falls back to the full size.
+ *
+ * Source: https://github.com/WordPress/gutenberg/blob/HEAD/packages/editor/src/components/post-featured-image/index.js
+ *
+ * @param {object} media
+ * @param {number} currentPostId
+ * @returns {?string} URL address
+ */
+export function getMediaSourceUrl( media, currentPostId ) {
+	if ( ! media ) {
+		return null;
+	}
+
+	const mediaSize = applyFilters(
+		'editor.PostFeaturedImage.imageSize',
+		'post-thumbnail',
+		media.id,
+		currentPostId
+	);
+	if ( has( media, [ 'media_details', 'sizes', mediaSize ] ) ) {
```

Comment:
Any benefit to `has` here over https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining?

Reply:
just in case...
![image](https://user-images.githubusercontent.com/77539/89529052-72ed4100-d7c2-11ea-9e99-d81f9b88cb6b.png)

Reply:
Updated, our code is compiled to be fully compatible even with IE